src/decorators.ts

```diff
@@ -71,42 +69,16 @@ export interface PropertyOptions {
   observer?: string|((val: {}, old: {}) => void);
 }
 
-interface Reflect {
-  hasMetadata?:
-      (metadataKey: string, proto: object, targetKey: string) => boolean;
-  getMetadata?: (metadataKey: string, proto: object, targetKey: string) =>
-      object | undefined;
-}
-
 function createProperty(
-    proto: ElementPrototype, name: string, options?: PropertyOptions): void {
+    proto: ElementPrototype, name: string, options?: Partial<PropertyOptions>):
```

Comment:
because we use `Partial` here, does that mean the required `type` further up becomes optional again?

Reply:
No, `property` will still enforce the narrower type, it's `Partial` here because `computed` doesn't set `type`.